pyomo/contrib/solver/solvers/gurobi_persistent.py

```diff
@@ -306,24 +282,16 @@ def release_license(self):
 
     def __del__(self):
         if not python_is_shutting_down():
-            Gurobi._num_instances -= 1
-            if Gurobi._num_instances == 0:
+            GurobiPersistent._num_instances -= 1
+            if GurobiPersistent._num_instances == 0:
```

Comment:
should `_num_instances` live on the mixin class and be shared by direct and persistent???

Reply:
I have been thinking about this question all weekend, and the answer I keep coming to is, "I'm not sure." Going back to the conversation with John last week, is that too much like passing around data (has-a vs. is-a)? I am fine with it either way (because functionally it works fine to put it in the mixin).

Reply:
Maybe some of this will go away. I think we need to discuss `acquire_license` and `release_license` more. Maybe these should be independent of the number of instances of solver interfaces. Right now, if a someone has instances of both direct and persistent interfaces, the `__del__` method of one could affect the other.